docs/learn/multi-format-reader.md

```diff
@@ -0,0 +1,263 @@
+    return template
+```
+As a last step, the template is being filled from the config dict using the data. If there is more than one entry, the `get_entry_names` method must be implemented, which shall return a list of all entry names. The `fill_from_config` method iterates through all of the them and replaces the generic `/ENTRY/` in the config file by keys of the form `/ENTRY[my-entry]/` to fill the template.
+
+Here, we are using **callbacks**, which are used to bring in data based on `@`-prefixes in the config file. These are defined in the reader's ``__init__`` call using the `pynxtools.dataconverter.readers.multi.ParseJsonCallbacks` class:
+```python
+self.callbacks = ParseJsonCallbacks(
+    attrs_callback=self.get_attr,
+    data_callback=self.get_data,
+    eln_callback=self.get_eln_data,
+    dims=self.get_data_dims,
+)
+```
+The `ParseJsonCallbacks` class has an attribute called `special_key_map` that makes use of these callbacks to populate the template based on the starting prefix of the config dict value:
+```python
+self.special_key_map = {
+    "@attrs": attrs_callback if attrs_callback is not None else self.identity,
+    "@link": link_callback if link_callback is not None else self.link_callback,
+    "@data": data_callback if data_callback is not None else self.identity,
+    "@eln": eln_callback if eln_callback is not None else self.identity,
+}
+```
+That means, if the config file has an entry ```{"/ENTRY/title": "@attrs:metadata/title"}```, the `get_attr` method of the reader gets called and should return an attribute from the given path, i.e., in this case from `metadata/title`.
+
+By default, the MultiFormatReader supports the following special prefixes:
+
+- `@attrs`: To get metadata from the read-in experiment file(s). You need to implement the `get_attr` method in the reader.
+- `@data`: To get measurement data from the read-in experiment file(s). You need to implement the `get_data` method in the reader.
+- `@eln`: To get metadata from addtional ELN files. You need to implement the `get_eln_data` method in the reader.
+- `@link`: To implement a link between two entities in the NeXus file. By default, the link callback returns a dict of the form {"link": value.replace("/entry/", f"/{self.entry_name}/")}, i.e., a generic `/entry/` get replaced by the actual `entry_name`.
+
+In addition, the reader can also implement the `get_data_dims` method, which is used to return a list of the data dimensions (see below for more details).
+
+All of `get_attr`, `get_data`, and `get_eln_data`  (as well as any similar method that might be implemented) should have the same call signature:
+```python
+def get_data(self, key: str, path: str) -> Any:
+```
+Here, `key` is the config dict key (e.g., `"/ENTRY[my-entry]/data/data"`) and path is the path that comes _after_ the prefix in the config file. In this example, `path` would be `mydata`. With these two inputs, the reader should be able to return the correct data for this template key.

```

Comment:
Not sure which example you refer to here with mydata (?)

Reply:
i was refering to the example config file given above. Clarified here.